src/fusion_algorithm.cpp

```diff
@@ -0,0 +1,325 @@
+
+/** @file fusion_algorithm.cpp
+ *  @brief contains the implementation of the sensor fusion algorithm
+ */
+
+// All C++ Headers
+#include "../inc/sensor_fusion_common.hpp"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// All C Headers go here
+
+#ifdef __cplusplus
+}
+#endif
+
+SensorsList_t remove_sensor_by_name(SensorsList_t sensors, char name[SENSOR_MAX_NAME_LEN]) {
```

Comment:
Here make a copy of the complete list of sensors. deleting from that copy. then send back another copy. This code highly inefficient, I will investigate. This should be a void function that is passed a pointer the sensor list. I will make the necessary changes.